src/test/scala/chiselTests/reflect/DataMirrorSpec.scala

```diff
@@ -245,4 +245,31 @@ class DataMirrorSpec extends ChiselFlatSpec {
     DataMirror.getLayerColor(foo.c) should be(Some(A))
   }
 
+  "moduleIOs" should "return an in-progress module's IOs" in {
+    class Foo extends RawModule {
+      val in = IO(Input(Bool()))
+      val out = IO(Output(Bool()))
+      val wire = Wire(Bool())
+      val child = Module(new RawModule {})
+
+      val ports0 = DataMirror.internal.currentModulePorts(this)
+
+      val other = IO(Input(Bool()))
+
+      val ports1 = DataMirror.internal.currentModulePorts(this)
+    }
+
+    ChiselStage.emitCHIRRTL(new RawModule {
+      val foo = Module(new Foo)
+
+      foo.ports0.size should be(2)
+      foo.ports0(0).toNamed.name should be("in")
+      foo.ports0(1).toNamed.name should be("out")
```

Comment:
```suggestion
      foo.ports0(0).toNamed.name should be("in")
      foo.ports0(1).toNamed.name should be("out")
```
Could this just be `foo.ports0 should be(Seq(foo.in, foo.out))`?